src/lib/storages/RedisProxyCache.ts

```diff
@@ -181,11 +246,60 @@ export class RedisProxyCache implements IProxyCache {
     }
   }
 
+  private async processNode(node: Redis, options: ProcessNodeOptions): Promise<void> {
+    const { pattern: match, batchSize: count, callbackFn, resolve, reject } = options
+    const stream = node.scanStream({ match, count })
+
+    stream.on('data', async (keys) => {
+      stream.pause()
+      try {
+        await Promise.all(keys.map((key: string) => this.processKey(key, callbackFn)))
+      } catch (err: unknown) {
+        stream.destroy(err as Error)
+        reject(err)
+      }
+      stream.resume()
+    })
+    stream.on('end', resolve)
+  }
+
+  private async processKey(key: string, callbackFn: ProcessKeyCallback): Promise<any> {
+    const actualKey = key.replace(':expiresAt', '')
+    const expiresAt = await this.redisClient.get(key)
+
+    if (Number(expiresAt) >= Date.now()) return
+    
+    try {
+      await callbackFn(actualKey)
+      if (this.isCluster) {
+        return Promise.all([
+          this.redisClient.del(actualKey),
+          this.redisClient.del(key)
+        ])
+      }
+      return this.executePipeline([
+        ['del', actualKey],
+        ['del', key]
+      ])
+    } catch (err: unknown) {
+      /**
+       * We don't want to throw an error here, as it will stop the whole process
+       * and we want to continue with the next keys
+       * We, however, need to decide on how/when to finally give up on a key and remove it from the cache
+       */
+      this.log.error('processKey error', err)
```

Comment:
i think, it might be useful to add key to log, e.g.:
```
this.log.error('error on processKey ${key} ', err)
```

Reply:
Added.